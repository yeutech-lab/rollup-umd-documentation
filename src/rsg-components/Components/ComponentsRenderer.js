import React from 'react';
import PropTypes from 'prop-types';

export default function ComponentsRenderer({ children }) {
	return <div>{children}</div>;
}
ComponentsRenderer.propTypes = {
	children: PropTypes.node.isRequired,
};
