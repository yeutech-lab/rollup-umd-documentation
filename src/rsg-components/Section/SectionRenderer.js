import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import SectionHeading from 'rsg-components/SectionHeading';
import Markdown from 'rsg-components/Markdown';

const styles = ({ space }) => ({
	root: {
		marginBottom: space[4],
	},
});

export function SectionRenderer(allProps) {
	const { classes, name, slug, content, components, sections, depth, description } = allProps;

	return (
		<section className={classes.root}>
			{name && (
				<SectionHeading depth={depth} id={slug} slotName="sectionToolbar" slotProps={allProps}>
					{name}
				</SectionHeading>
			)}
			{description && <Markdown text={description} />}
			{content}
			{sections}
			{components}
		</section>
	);
}

SectionRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	name: PropTypes.string,
	description: PropTypes.string,
	slug: PropTypes.string,
	filepath: PropTypes.string,
	content: PropTypes.node,
	components: PropTypes.node,
	sections: PropTypes.node,
	isolated: PropTypes.bool,
	depth: PropTypes.number.isRequired,
};

export default Styled(styles)(SectionRenderer);
