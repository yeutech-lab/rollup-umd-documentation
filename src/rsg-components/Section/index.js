export { default } from 'rsg-components/Section/Section';
