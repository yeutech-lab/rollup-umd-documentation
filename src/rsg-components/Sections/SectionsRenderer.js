import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

const styles = () => ({
	// Just default jss-isolate rules
	root: {},
});

export function SectionsRenderer({ classes, children }) {
	return <section className={classes.root}>{children}</section>;
}

SectionsRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.node,
};

export default Styled(styles)(SectionsRenderer);
