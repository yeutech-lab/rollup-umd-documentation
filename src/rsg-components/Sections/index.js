export { default } from '../Sections/Sections';
