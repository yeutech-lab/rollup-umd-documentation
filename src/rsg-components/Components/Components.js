import React from 'react';
import PropTypes from 'prop-types';
import ReactComponent from 'rsg-components/ReactComponent';
import ComponentsRenderer from 'rsg-components/Components/ComponentsRenderer';

export default function Components({ components, depth }) {
	return (
		<ComponentsRenderer>
			{components.map(component => (
				<ReactComponent key={component.filepath} component={component} depth={depth} />
			))}
		</ComponentsRenderer>
	);
}

Components.propTypes = {
	components: PropTypes.array.isRequired,
	depth: PropTypes.number.isRequired,
};
