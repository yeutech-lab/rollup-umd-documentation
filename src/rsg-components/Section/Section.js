import React from 'react';
import PropTypes from 'prop-types';
import Examples from 'rsg-components/Examples';
import Components from 'rsg-components/Components';
import Sections from 'rsg-components/Sections';
import SectionRenderer from 'rsg-components/Section/SectionRenderer';
import { DisplayModes } from '../../consts';

export default function Section({ section, depth }, { displayMode }) {
	const { name, slug, filepath, content, components, sections, description } = section;

	const contentJsx = content && <Examples examples={content} name={name} />;
	const componentsJsx = components && <Components components={components} depth={depth + 1} />;
	const sectionsJsx = sections && <Sections sections={sections} depth={depth + 1} />;

	return (
		<SectionRenderer
			description={description}
			name={name}
			slug={slug}
			filepath={filepath}
			content={contentJsx}
			components={componentsJsx}
			sections={sectionsJsx}
			isolated={displayMode !== DisplayModes.all}
			depth={depth}
		/>
	);
}

Section.propTypes = {
	section: PropTypes.object.isRequired,
	depth: PropTypes.number.isRequired,
};

Section.contextTypes = {
	displayMode: PropTypes.string,
};
