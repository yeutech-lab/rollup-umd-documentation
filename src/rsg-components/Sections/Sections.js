import React from 'react';
import PropTypes from 'prop-types';
import Section from 'rsg-components/Section';
import SectionsRenderer from 'rsg-components/Sections/SectionsRenderer';

export default function Sections({ sections, depth }) {
	return (
		<SectionsRenderer>
			{sections.map((section, idx) => <Section key={idx} section={section} depth={depth} />)}
		</SectionsRenderer>
	);
}

Sections.propTypes = {
	sections: PropTypes.array.isRequired,
	root: PropTypes.bool,
	depth: PropTypes.number.isRequired,
};
