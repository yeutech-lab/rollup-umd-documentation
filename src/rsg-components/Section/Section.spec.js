import React from 'react';
import noop from 'lodash/noop';
import Examples from '../Examples';
import Components from '../Components';
import Sections from '../Sections';
import Section from './Section';
import { SectionRenderer } from './SectionRenderer';
import { DisplayModes } from '../../consts';

const options = {
	context: {
		displayMode: DisplayModes.all,
	},
};

const section = {
	name: 'Foo',
	slug: 'foo',
	description: 'This is a description',
	content: [
		{
			type: 'code',
			content: '<button>OK</button>',
			evalInContext: noop,
		},
		{
			type: 'markdown',
			content: 'Hello *world*!',
		},
	],
	components: [],
	sections: [],
};

it('should render section renderer', () => {
	const actual = shallow(<Section section={section} depth={3} />, options);

	expect(actual).toMatchSnapshot();
});

it('should render components list', () => {
	const actual = shallow(
		<Section
			section={{
				name: 'Components',
				slug: 'components',
				components: [],
			}}
			depth={3}
		/>,
		options
	);

	expect(actual).toMatchSnapshot();
});

it('should not render components list if not defined', () => {
	const actual = shallow(
		<Section
			section={{
				name: 'No components',
				slug: 'no-components',
			}}
			depth={3}
		/>,
		options
	);

	expect(actual).toMatchSnapshot();
});

it('should render sections if defined', () => {
	const actual = shallow(
		<Section
			section={{
				name: 'Nested sections',
				slug: 'nested-sections',
				sections: [],
			}}
			depth={3}
		/>,
		options
	);

	expect(actual).toMatchSnapshot();
});

it('should not render sections if not defined', () => {
	const actual = shallow(
		<Section
			section={{
				name: 'No sections',
				slug: 'no-sections',
			}}
			depth={3}
		/>,
		options
	);

	expect(actual).toMatchSnapshot();
});

test('should not render section in isolation mode by default', () => {
	const actual = shallow(
		<Section
			section={{
				name: 'A',
				slug: 'a',
			}}
			depth={3}
		/>,
		options
	);

	expect(actual.prop('isolated')).toBeFalsy();
});

test('should render section in isolation mode', () => {
	const actual = shallow(
		<Section
			section={{
				name: 'A',
				slug: 'a',
			}}
			depth={3}
		/>,
		{
			context: {
				...options.context,
				displayMode: DisplayModes.section,
			},
		}
	);

	expect(actual.prop('isolated')).toBeTruthy();
});

it('render should render section', () => {
	const actual = shallow(
		<SectionRenderer
			classes={{}}
			name={section.name}
			slug={section.slug}
			content={<Examples name={section.name} examples={section.content} />}
			components={<Components components={[]} depth={3} />}
			sections={<Sections sections={[]} depth={3} />}
			depth={3}
		/>,
		options
	);

	expect(actual).toMatchSnapshot();
});

it('render should not render title if name is not set', () => {
	const actual = shallow(<SectionRenderer classes={{}} depth={3} />, options);

	expect(actual).toMatchSnapshot();
});

it('render should render title if name is set', () => {
	const actual = shallow(
		<SectionRenderer classes={{}} name="test" slug="test" depth={3} />,
		options
	);

	expect(actual).toMatchSnapshot();
});
