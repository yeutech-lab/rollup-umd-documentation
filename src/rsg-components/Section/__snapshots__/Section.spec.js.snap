// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should not render title if name is not set 1`] = `<section />`;

exports[`render should render section 1`] = `
<section>
  <SectionHeading
    depth={3}
    id="foo"
    slotName="sectionToolbar"
    slotProps={
      Object {
        "classes": Object {},
        "components": <Components
          components={Array []}
          depth={3}
        />,
        "content": <Examples
          examples={
            Array [
              Object {
                "content": 
<button>
  OK
</button>
,
                "evalInContext": [Function],
                "type": "code",
              },
              Object {
                "content": "Hello *world*!",
                "type": "markdown",
              },
            ]
          }
          name="Foo"
        />,
        "depth": 3,
        "name": "Foo",
        "sections": <Sections
          depth={3}
          sections={Array []}
        />,
        "slug": "foo",
      }
    }
  >
    Foo
  </SectionHeading>
  <Examples
    examples={
      Array [
        Object {
          "content": 
<button>
  OK
</button>
,
          "evalInContext": [Function],
          "type": "code",
        },
        Object {
          "content": "Hello *world*!",
          "type": "markdown",
        },
      ]
    }
    name="Foo"
  />
  <Sections
    depth={3}
    sections={Array []}
  />
  <Components
    components={Array []}
    depth={3}
  />
</section>
`;

exports[`render should render title if name is set 1`] = `
<section>
  <SectionHeading
    depth={3}
    id="test"
    slotName="sectionToolbar"
    slotProps={
      Object {
        "classes": Object {},
        "depth": 3,
        "name": "test",
        "slug": "test",
      }
    }
  >
    test
  </SectionHeading>
</section>
`;

exports[`should not render components list if not defined 1`] = `
<Styled(Section)
  depth={3}
  isolated={false}
  name="No components"
  slug="no-components"
/>
`;

exports[`should not render sections if not defined 1`] = `
<Styled(Section)
  depth={3}
  isolated={false}
  name="No sections"
  slug="no-sections"
/>
`;

exports[`should render components list 1`] = `
<Styled(Section)
  components={
    <Components
      components={Array []}
      depth={4}
    />
  }
  depth={3}
  isolated={false}
  name="Components"
  slug="components"
/>
`;

exports[`should render section renderer 1`] = `
<Styled(Section)
  components={
    <Components
      components={Array []}
      depth={4}
    />
  }
  content={
    <Examples
      examples={
        Array [
          Object {
            "content": 
<button>
  OK
</button>
,
            "evalInContext": [Function],
            "type": "code",
          },
          Object {
            "content": "Hello *world*!",
            "type": "markdown",
          },
        ]
      }
      name="Foo"
    />
  }
  depth={3}
  description="This is a description"
  isolated={false}
  name="Foo"
  sections={
    <Sections
      depth={4}
      sections={Array []}
    />
  }
  slug="foo"
/>
`;

exports[`should render sections if defined 1`] = `
<Styled(Section)
  depth={3}
  isolated={false}
  name="Nested sections"
  sections={
    <Sections
      depth={4}
      sections={Array []}
    />
  }
  slug="nested-sections"
/>
`;
